packages/util-valid-hostname/src/index.ts

```diff
@@ -0,0 +1,4 @@
+export function validHostname(hostname: string): boolean {
+  const hostPattern = /^[a-zA-Z0-9]{1}$|^[a-zA-Z0-9][a-zA-Z0-9\-]*[a-zA-Z0-9]$/;
```

Comment:
Shouldn't you compile this once and then reuse the regex?

Reply:
Using the literal (vs the RegExp constructor) should cache the regex:
```
const { validHostname } = require("/aws-sdk-js-v3/packages/util-valid-hostname");

console.time('1');validHostname("test1");console.timeEnd('1');
console.time('2');validHostname("test2");console.timeEnd('2');
```
Node returns:
```
1: 0.262ms
2: 0.011ms
```

Reply:
Remove standalone package and added function to protocol-http.